nginx-controller/controller/controller.go

```diff
@@ -673,20 +673,60 @@ func (lbc *LoadBalancerController) syncIng(task Task) {
 
 	if !ingExists {
 		glog.V(2).Infof("Deleting Ingress: %v\n", key)
+
+		// TODO(diazjf): Wont be able to delete a minion and will cause runtime error,
```

Comment:
In the delete function, it tries to delete a file for an Ingress Resource. In the case of a minion, it will not exist. Since we are not getting all the resources and only processing Minions when a master exists, the removal case won't instantly apply until another minion or a master is updated.

I guess what we can do in this case is just look through all the ingress resources and if a master exists, look for all its minions and update. We would do this every time since we don't know what it deleted.

Also should we have .minion files which get configured/removed? How would we stop a runtime error when a minion resource is trying to be deleted.

Reply:
@diazjf I suggest solving this problem by changing  [`ingHandlers`](https://github.com/diazjf/kubernetes-ingress/blob/31fc818250592cd4438a9aec05fce18051ea8c04/nginx-controller/controller/controller.go#L97). In `AddFunc`, `DeleteFunc` and `UpdateFunc`, if you encounter a minion, find its master and then enqueue the master. In this case, if a minion is deleted, the master will be added to a queue, which will lead to regenerating the config and thus deleting the minion from the config.

Reply:
Perfect!